### PR TITLE
Reject execution of empty test sets

### LINKED ISSUE #674: Add backend validation to prevent execution of empty test sets
## Summary
Add backend validation to prevent execution of test sets with 0 tests, complementing the frontend UI fix in PR #651.

## Background / Context
PR #651 adds a frontend fix that disables the "Execute Test Set" button when a test set has 0 tests. However, the backend still accepts execution requests for empty test sets, which creates unnecessary worker tasks and provides a poor user experience. Related to issue #592.

## Goals
- Add backend validation to reject execution attempts for empty test sets
- Return clear error messages explaining why execution was rejected
- Prevent unnecessary worker tasks from being created
- Maintain consistency between frontend and backend validation

## Deliverables
1. Add validation check in `execute_test_set_on_endpoint()` function
2. Add validation check in worker task that executes test configurations
3. Return HTTP 400 with descriptive error message
4. Add backend tests to verify empty test set rejection
5. Update API documentation if needed

### Backend API Layer
In `apps/backend/src/rhesis/backend/app/services/test_set.py`:
1. Add validation after resolving test set in `execute_test_set_on_endpoint()` (around line 519)
2. Check if test set has `total_tests == 0` or no tests associated
3. Raise `ValueError` with clear message: "Cannot execute test set with 0 tests"

### Worker Layer
In the test execution worker task:
1. Add similar validation as a safety check before execution
2. Fail task gracefully with descriptive error if empty test set detected

### Tests
Add test cases in `tests/backend/routes/test_test_set.py`:
1. Test attempting to execute empty test set returns 400 error
2. Test error message is descriptive
3. Test that execution succeeds for test sets with at least 1 test

## Acceptance Criteria
- [ ] Attempting to execute a test set with 0 tests returns HTTP 400 Bad Request
- [ ] Error message clearly states: "Cannot execute test set with 0 tests" or similar
- [ ] Worker task validates test count and fails gracefully if empty
- [ ] No Celery tasks are created for empty test set execution attempts
- [ ] Backend tests verify empty test set rejection
- [ ] Test sets with 1+ tests can still be executed successfully

## Dependencies
PR #651 should be merged first as it provides the frontend validation.

## Risks / Considerations
- Need to check how `total_tests` is calculated/stored in test set metadata
- Must ensure validation happens early to avoid creating unnecessary database records
- Worker validation should handle edge cases (e.g., tests deleted after configuration created)

## Additional Context
Related to PR #651 (frontend fix) and issue #592 (original issue).

Current execution flow:
1. Frontend → `POST /test_sets/{id}/execute/{endpoint_id}`
2. Backend creates test configuration
3. Backend submits Celery task
4. Worker executes tests

Validation should happen at step 2 to prevent steps 3-4 for empty test sets.

Example validation code:
```python
# In execute_test_set_on_endpoint(), after line 519:
if db_test_set.total_tests == 0:
    raise ValueError("Cannot execute test set with 0 tests. Please add tests before executing.")
```

### PR DESCRIPTION
**Pull request by Aawegg**
## Summary
- reject empty test sets before creating execution configuration or scheduling Celery work
- count only active tests, excluding stale associations to soft-deleted records
- retain a worker-side safety check if tests are removed after the request is accepted
- add service and orchestration coverage, including a soft-delete regression test

Closes #674

## Test plan
- [x] `ruff check` on all changed files
- [x] `ruff format --check` on changed service and test files
- [x] `python -m py_compile` on changed files
- [ ] backend integration tests run in project CI because local Docker was unavailable

### CODE REVIEW

**Comment by nicolai-rhesis** [comment]
Thanks for the contribution — this looks like a solid fix for #674, and the dual validation (API before scheduling + worker-side safety check) with tests is exactly what we were looking for. Really appreciate you picking this up as a first-time contributor!

A couple of small asks before we merge:

1. **DCO sign-off** — We use the [Developer Certificate of Origin (DCO)](https://developercertificate.org/) and ask contributors to sign off on their commits (see the [Legal Notice in CONTRIBUTING.md](https://github.com/rhesis-ai/rhesis/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-dco)). Could you re-commit with the `-s` flag so the commit includes a `Signed-off-by` line? For example:

```bash
git commit --amend -s --no-edit
git push --force-with-lease
```

2. **PEQY review notes** — Our review bot left a couple of improvement suggestions on the PR (soft-deleted tests can leave association rows behind, so `count_test_set_tests()` may disagree with `test_set.tests`). Could you kindly take a look and address those if you agree they make sense?

Thanks again!

File: apps/backend/src/rhesis/backend/app/services/test_set.py [hunk 1]
```diff
@@ -847,6 +850,22 @@ def execute_test_set_on_endpoint(
     return response_data
 
 
+def count_test_set_tests(db: Session, test_set_id: uuid.UUID) -> int:
+    """Count tests associated with a test set via the association table."""
+    return (
```
[the remark this answers is not in the document]
Comment (Aawegg) [reply apps/backend/src/rhesis/backend/app/services/test_set.py:859]: Addressed in 848ef38: `count_test_set_tests()` now joins `models.Test` and explicitly filters `Test.deleted_at IS NULL`, so stale association rows no longer count. I also added a regression test covering an associated test before and after soft deletion.

File: apps/backend/src/rhesis/backend/tasks/execution/orchestration.py [hunk 2]
```diff
@@ -39,16 +40,12 @@ def execute_test_cases(
     """
 
     test_set = get_test_set(session, str(test_config.test_set_id))
-    tests = test_set.tests
+    if count_test_set_tests(session, test_set.id) == 0:
```
[the remark this answers is not in the document]
Comment (Aawegg) [reply apps/backend/src/rhesis/backend/tasks/execution/orchestration.py:43]: Addressed through the shared helper in 848ef38. The worker-side check now receives the active-test count, so a test set whose linked tests are all soft-deleted raises `TestExecutionError` instead of proceeding with an empty relationship.

**Comment by Aawegg** [comment]
Thanks for the review. I addressed both asks in 848ef38:

- added the DCO `Signed-off-by` line to the sole commit
- changed the helper to join `Test` and count only rows with `deleted_at IS NULL`
- added a regression test proving a stale association to a soft-deleted test counts as zero

Lint, formatting, and syntax checks pass locally; CI has restarted on the updated commit.
